src/main/java/com/palantir/gradle/revapi/OldApiConfigurations.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.revapi;
+
+import com.palantir.gradle.revapi.config.Version;
+import java.io.File;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.result.DependencyResult;
+import org.gradle.api.artifacts.result.UnresolvedDependencyResult;
+
+final class OldApiConfigurations {
+    private OldApiConfigurations() {}
+
+    static Configuration configuration(
+            Project project,
+            Dependency oldApiDependency,
+            String name,
+            String description) {
+
+        return project.getConfigurations().create(name, conf -> {
+            conf.setDescription(description);
+            conf.getDependencies().add(oldApiDependency);
+            conf.setCanBeConsumed(false);
+            conf.setVisible(false);
+        });
+    }
+
+    static Set<File> resolveConfigurationUnlessMissingJars(Version oldVersion, Configuration configuration)
+            throws CouldNotResolveOldApiException {
+
+        Set<? extends DependencyResult> allDependencies = configuration.getIncoming()
+                .getResolutionResult()
+                .getAllDependencies();
+
+        List<Throwable> resolutionFailures = allDependencies.stream()
+                .filter(dependencyResult -> dependencyResult instanceof UnresolvedDependencyResult)
+                .map(dependencyResult -> (UnresolvedDependencyResult) dependencyResult)
+                .map(UnresolvedDependencyResult::getFailure)
+                .collect(Collectors.toList());
+
+        if (resolutionFailures.isEmpty()) {
+            return configuration.resolve();
+        }
+
+        throw new CouldNotResolveOldApiException(oldVersion, resolutionFailures);
```

Comment:
it's weird that we're passing in this `oldVersion` only to put it back into an exception, so it doesn't actually affect the behaviour of this method).

Perhaps we could construct the configuration in here, and have the function take a name/prefix for the configuration?

essentially putting this code https://github.com/palantir/gradle-revapi/pull/108/files#diff-1da680adc77435cdeb6de908e5415d62R84-R90 inside the function.

Reply:
I'm still passing in `oldVersion` here, but this method is now private, so I don't think it's as bad (and inlining this method would look ugly, I think).